juniper_official/vpn/l3vpn/l3vpn-proto-bgp-nw.rule

```diff
@@ -0,0 +1,173 @@
+                constant {
+                    value "{{customer-vpn-name}}";
+                }
+                type string;
+                description "VRF name";
+            }
+            field vpn-state {
+                reference {
+                    path "/device-group[device-group-name={{pe-device-group}}]/device[device-id={{pe-device-name}}]/topic[topic-name='routing.bgp']/rule[rule-name=check-bgp-neighbor-stats]/field[peer-address='{{customer-neighbor-address}}']/peer-state";
+                }
+                type string;
+                description "VRF peer state";
+            }
+            trigger l3vpn-pe-interface-state {
+                frequency 1offset;
+                term is-pe-interface-up {
+                    when {
+                        matches-with "$instance-interface-status" UP;
+                    }
+                    then {
+                        status {
+                            color green;
+                            message "PE interface $instance-interface-name.$instance-ifl-no of VPN:$vpn-name on router $pe-router-name is UP";
+                        }
+                    }
+                }
+                term is-pe-interface-down {
+                    then {
+                        status {
+                            color red;
+                            message "PE interface $instance-interface-name.$instance-ifl-no of VPN:$vpn-name on router $pe-router-name is $instance-interface-status";
+                        }
+                    }
+                }
+            }
+            trigger l3vpn-state {
+                synopsis "L3VPN session status KPI";
+                description "Sets health based on state change in L3VPN peer and PE interface state.";
+                frequency 1offset;
+                term is-vpn-session-up {
+                    when {
+                        matches-with "$vpn-state" ESTABLISHED;
+                    }
+                    then {
+                        status {
+                            color green;
+                            message "L3VPN:$vpn-name neighbor:$neighbor-session of $pe-router-name is UP";
+                        }
+                    }
+                }
+                term is-vpn-session-down {
+                    then {
+                        status {
+                            color red;
+                            message "L3VPN:$vpn-name neighbor:$neighbor-session of $pe-router-name is $vpn-state";
+                        }
+                    }
+                }
+            }
+            variable customer-neighbor-address {
+                description "BGP neighbor addresses to monitor, regular expression, e.g. '172.16.*'";
+                type string;
+            }
+            /*
+             * Variables with default values. Default values can be
+             * changed while deploying playbook instance.
+             */
+            variable customer-vpn-name {
```

Comment:
Why can't all the three network rules be combined. I see same inputs to all 3 rules except address in one of the rule. Lets try to optimise the processing and storage as we have control on the use cases now.

Reply:
There will be only one type of routing will be selected between PE and CE from ospf/bgp/static hence these 3 rules should be seperate.